pkg/kclient/kclient.go

```diff
@@ -0,0 +1,53 @@
+package kclient
+
+import (
+	"errors"
+
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// errorMsg is the message for user when invalid configuration error occurs
+const errorMsg = `
```

Comment:
That's definitely an interesting approach, and at that article definitely has some merits. But I'm wondering if we want to implement it as part of our `errors` package we have or maybe a subpackage in kclient if we want to isolate things (and to prevent having to update a ton of existing odo code)

My concern is around ensuring we can use the existing functions in our `errors` package, and defining a custom error type might cause problems with currently.